nemo/collections/nlp/data/language_modeling/megatron/megatron_dataset.py

```diff
@@ -46,3 +46,13 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
                 tensor_model_parallel_size=cfg.get('tensor_model_parallel_size', 1),
                 seed=self.cfg.get('seed', 1234),
             )
+
+        try:
+            from nemo.collections.nlp.data.language_modeling.megatron.dataset_utils import compile_helper
+
+            compile_helper()
+            logging.info('Megatron dataset helper compiled successfully.')
+            from nemo.collections.nlp.data.language_modeling.megatron import helpers
+        except ImportError:
+            logging.error('Could not compile megatron dataset C++ helper functions.')
+            raise ImportError(f'Could not compile megatron dataset C++ helper functions and so cannot import helpers.')
```

Comment:
Do we need both `logging.error` and `raise ImportError` here?

Reply:
Good point, the exception will be logged, we could probably remove the log.error (unless we want to continue run when dataset compilation fails)

Reply:
Cool, I'll just raise the ImportError and leave it at that then.